### ISSUE TITLE
Waterfall.blank_dc() causes numpy "RuntimeWarning: Mean of empty slice"

### ISSUE THREAD

**Issue by texadactyl** [UCBerkeleySETI/blimpy#212]
In some cases, blank_dc().parse() causes numpy to emit messages:
"RuntimeWarning: Mean of empty slice"

Context:
        def parse(data, n_coarse_chan, n_chan_per_coarse, mid_chan):
            for ii in range(n_coarse_chan):
                ss = ii*n_chan_per_coarse
                data[..., ss+mid_chan] = np.median(data[..., ss+mid_chan+5:ss+mid_chan+10]) # !!!!!

Fortunately, it seems like it causes no damage - the action desired just does not happen.  I cannot comment on the resulting effects of the smoothing over not taking place.

@telegraphic ?

**Comment by texadactyl** [comment]
Higher context:
blank_dc() is being called by turbo_seti find_doppler.py:load_data().  Necessary?

**Comment by telegraphic** [comment]
Hmm, well the DC bin is often very large in amplitude, so can affect statistics (although it looks like we're already using median in `comp_stats`, which you noted elsewhere). So it's a good idea to blank it -- even though all the poorly defined coarse channel stuff is an albatross. 

This seems to suggest that the `np.median` is being passed an empty slice, i.e. that the logic behind `ss+mid_chan` stuff is going wrong.

**Comment by texadactyl** [comment]
Agreed.  Unfortunately, the code was written in such a way that it is hard to trap or put print-debug statements to confirm.  I will look at this as time permits after my cryptography obligations today.

**Comment by texadactyl** [comment]
Note this blank_dc middle portion of the code:
```
        n_coarse_chan = int(n_coarse_chan)
        n_chan = self.data.shape[-1]
        n_chan_per_coarse = int(n_chan / n_coarse_chan)
```
When the number of fine channels per coarse channel (```n_chan_per_coarse```) < 5 (E.g. some MeerKAT files = 4), then an empty slice can be generated.

**Comment by texadactyl** [comment]
Fixed in PR #217
